ddht/tools/driver/alexandria.py

```diff
@@ -0,0 +1,74 @@
+from typing import AsyncContextManager, AsyncIterator, Optional
+
+from async_generator import asynccontextmanager
+from async_service import background_trio_service
+import trio
+
+from ddht._utils import asyncnullcontext
+from ddht.tools.driver.abc import AlexandriaNodeAPI, NodeAPI
+from ddht.v5_1.abc import NetworkAPI
+from ddht.v5_1.alexandria.abc import AlexandriaClientAPI, AlexandriaNetworkAPI
+from ddht.v5_1.alexandria.client import AlexandriaClient
+from ddht.v5_1.alexandria.network import AlexandriaNetwork
+
+
+class AlexandriaNode(AlexandriaNodeAPI):
+    _listen_lock: trio.Lock
+    _lock_owner: Optional[str]
+
+    def __init__(self, node: NodeAPI) -> None:
+        self.node = node
+        self._listen_lock = trio.Lock()
+        self._lock_owner = None
+
+    @asynccontextmanager
+    async def _acquire_listen_lock(self, owner: str) -> AsyncIterator[None]:
```

Comment:
A comment would be useful here, I'm not sure what this lock is meant to protect. I think `self.node.network()`?